src/core/Akka.Tests/Pattern/CircuitBreakerSpec.cs

```diff
@@ -7,174 +7,137 @@
+            breaker.Instance.CurrentFailureCount.ShouldBe(0);
+            InterceptException<TestException>(() => breaker.Instance.WithSyncCircuitBreaker(ThrowException));
+            breaker.Instance.CurrentFailureCount.ShouldBe(1);
+            breaker.Instance.WithSyncCircuitBreaker(SayHi);
+            breaker.Instance.CurrentFailureCount.ShouldBe(0);
         }
 
-        [Fact(DisplayName = "A synchronous circuit breaker that is closed must reset failure count and clears cached last exception after success method")]
+        [Fact(DisplayName = "A synchronous circuit breaker that is closed must reset failure count after success method")]
         public void Must_reset_failure_count_after_success_method()
         {
             var breaker = MultiFailureCb();
-            Assert.True(breaker.Instance.CurrentFailureCount == 0);
-            Assert.True(InterceptExceptionType<TestException>(() => breaker.Instance.WithSyncCircuitBreaker(ThrowException)));
-            Assert.True(breaker.Instance.CurrentFailureCount == 1);
-            Assert.True(breaker.Instance.LastCaughtException is TestException);
+            breaker.Instance.CurrentFailureCount.ShouldBe(0);
+            InterceptException<TestException>(() => breaker.Instance.WithSyncCircuitBreaker(ThrowException));
+            breaker.Instance.CurrentFailureCount.ShouldBe(1);
             breaker.Instance.Succeed();
-            Assert.True(breaker.Instance.CurrentFailureCount == 0);
-            Assert.True(breaker.Instance.LastCaughtException is null);
+            breaker.Instance.CurrentFailureCount.ShouldBe(0);
         }
-    }
 
-    public class ASynchronousCircuitBreakerThatIsHalfOpen : CircuitBreakerSpecBase
-    {
-        [Fact(DisplayName = "A synchronous circuit breaker that is half open should pass call and transition to close on success")]
-        public void Should_Pass_Call_And_Transition_To_Close_On_Success( )
+        [Fact(DisplayName = "A synchronous circuit breaker that is closed must increment failure count on callTimeout before call finishes")]
+        public void Must_increment_failure_count_on_callTimeout_before_call_finishes()
         {
-            var breaker = ShortResetTimeoutCb( );
-            InterceptExceptionType<TestException>( ( ) => breaker.Instance.WithSyncCircuitBreaker( ThrowException ) );
-            Assert.True( CheckLatch( breaker.HalfOpenLatch ) );
-
-            var result = breaker.Instance.WithSyncCircuitBreaker( ( ) => SayTest( ) );
-
-            Assert.True( CheckLatch( breaker.ClosedLatch ) );
-            Assert.Equal( SayTest( ), result );
+            var breaker = ShortCallTimeoutCb();
+            Task.Run(() => breaker.Instance.WithSyncCircuitBreaker(() => Thread.Sleep(Dilated(TimeSpan.FromSeconds(1)))));
+            Within(TimeSpan.FromMilliseconds(900),
+                () => AwaitCond(() => breaker.Instance.CurrentFailureCount == 1, Dilated(TimeSpan.FromMilliseconds(100))));
```

Comment:
I think your problem here is your parameters on `AwaitCond` - you meant to set `Dilated(TimeSpan.FromMilliseconds(100))` as the interval but it's being used as the max value here. @ismaelhamed

Reply:
https://github.com/akka/akka/blob/13892f7d01e2b0b0a4fbb7e20b3bed3bafee3dde/akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerSpec.scala#L413

Reply:
@Aaronontheweb , see my comment above. In the meantime, I've fixed it by passing both `max` and `interval`.